spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactory.java

```diff
@@ -115,8 +115,11 @@ private TopicPartition resolveDestination(ConsumerRecord<?, ?> cr, Exception e)
 
 		return nextDestination.isNoOpsTopic()
 					? null
-					: new TopicPartition(nextDestination.getDestinationName(),
-				cr.partition() % nextDestination.getDestinationPartitions());
+					: resolveTopicPartition(cr, nextDestination);
+	}
+
+	protected TopicPartition resolveTopicPartition(final ConsumerRecord<?, ?> cr, final DestinationTopic nextDestination) {
```

Comment:
Needs javadoc since it's a public API.

It should explain the default resolution algorithm.

Reply:
Added javadoc. Thanks!